### PR TITLE
cargo-tauri: 1.7.1-unstable-2024-08-16 -> 2.1.0

### PR DESCRIPTION
**Pull request by getchoo**
Closes #355468

Changelog: https://github.com/tauri-apps/tauri/releases/tag/tauri-v2.0.1
Diff: https://github.com/tauri-apps/tauri/compare/2b61447dfc167ec11724f99671bf9e2de0bf6768...tauri-v2.0.1

**This release contains breaking changes**. See the [migration guide](https://tauri.app/start/migrate/from-tauri-1/)

`cargo-tauri_1` has also been introduced in order to keep other packages that still haven't updated to v2 building. It should be removed in 25.05 as it will no longer be maintained upstream

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Eveeifyeve** [comment]
Also add a eval warning to Nix/NixOs.

**Comment by getchoo** [comment]
Nothing is removed, so I don't believe there is much of a need; I also wouldn't want to cause evaluation warnings for all packages in the tree that still use v1. The release notes should do well enough for now as breaking updates are expected in new releases, and then a `throw` can be used for `cargo-tauri_1` in 25.05 once it's no longer needed here

**Comment by Eveeifyeve** [comment]
It seems like @ofborg is failing since it missing maintainers.

File: pkgs/by-name/ca/cargo-tauri/package.nix
Comment (Eveeifyeve) [inline pkgs/by-name/ca/cargo-tauri/package.nix:33]:
```suggestion
cargoHash = "hash"
```

as this is not really needed as [`buildRustPackage` uses cargoLock to provide a hash](https://github.com/NixOS/nixpkgs/blob/5966581aa04be7eff830b9e1457d56dc70a0b798/pkgs/build-support/rust/build-rust-package/default.nix#L62-L79) instead to make it easier for your self use `cargohash`.

Comment (Eveeifyeve) [reply pkgs/by-name/ca/cargo-tauri/package.nix:33]: https://github.com/NixOS/nixpkgs/blob/5966581aa04be7eff830b9e1457d56dc70a0b798/pkgs/build-support/rust/import-cargo-lock.nix

Comment (getchoo) [reply pkgs/by-name/ca/cargo-tauri/package.nix:33]: I believe you're misunderstanding some things here. From the "Vendoring dependencies" [section](https://nixos.org/manual/nixpkgs/unstable/#vendoring-of-dependencies) section of the Rust reference:

> If the Cargo.lock file includes git dependencies, then their output hashes need to be specified since they are not available through the lock file.

We can't use `fetchCargoTarball`/`cargoHash` to do this at all

Comment (Eveeifyeve) [reply pkgs/by-name/ca/cargo-tauri/package.nix:33]: I guess it makes sense as nix is ment to be reproducible.

File: pkgs/by-name/ca/cargo-tauri/Cargo.lock
Comment (Eveeifyeve) [inline pkgs/by-name/ca/cargo-tauri/Cargo.lock:1]: Also can't you pull this from source?

File: pkgs/by-name/ca/cargo-tauri/package.nix
Comment (Eveeifyeve) [inline pkgs/by-name/ca/cargo-tauri/package.nix:29]:
```suggestion
    lockFile = ${src}/Cargo.lock;
```
As from comment https://github.com/NixOS/nixpkgs/pull/346277/files#r1788374298

Comment (Eveeifyeve) [reply pkgs/by-name/ca/cargo-tauri/package.nix:29]: This suggestion works.

Comment (getchoo) [reply pkgs/by-name/ca/cargo-tauri/package.nix:29]: This is IFD, which is [not allowed in nixpkgs](https://github.com/NixOS/nixpkgs/tree/92248b23e5925b7a69108488e1f6d4577eea9f76/pkgs#import-from-derivation)

Comment (Eveeifyeve) [reply pkgs/by-name/ca/cargo-tauri/package.nix:29]: Oh didn't know.

**Review by nim65s — CHANGES_REQUESTED** [review]
There is a merge conflict on master

**Comment by kilianar** [comment]
There’s currently a merge conflict with `master` due to recent changes: `kiwitalk` has been removed from `nixpkgs`, and both `insulator2` and `modrinth-app-unwrapped` have been updated.

**Comment by getchoo** [comment]
Merge conflicts have been fixed, `cargo-tauri` was bumped from 2.0.1 -> 2.0.6, `cargo-tauri-1` was bumped from 1.8.0 -> 1.8.1, and both should (hopefully) now build with the new Darwin stdenv (i.e., without the old explicit frameworks)

**Comment by gepbird** [comment]
There's a new merge conflict in the release notes

**Comment by HannesGitH** [comment]
hey, as i needed tauri 2 as well i forked your progress and started from there. its included in my pr: #355062 
which can be seen as a duplicate of this one, i just bumped the version further (2.1.0) and removed the uneccessary cargo.lock (as it can be taken from src) and wanted to share my progress (also, it has no conflicts ;))

**Comment by pyrox0** [comment]
Cargo.lock is vendored, because as [mentioned previously](https://github.com/NixOS/nixpkgs/pull/346277#discussion_r1788461518), doing `"${src}/Cargo.lock"` is a usage of import-from-derivation(IFD) and isn't allowed in nixpkgs. It's reasonable, in this case, to vendor Cargo.lock.

**Comment by getchoo** [comment]
The pnpmDeps hash also had to be updated. Thanks for the patch though, @HannesGitH!

**Comment by getchoo** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 346277`

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 3 packages marked as broken and skipped:</summary>
  <ul>
    <li>alcom</li>
    <li>modrinth-app</li>
    <li>modrinth-app-unwrapped</li>
  </ul>
</details>
<details>
  <summary>:fast_forward: 1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>gitbutler</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 9 packages built:</summary>
  <ul>
    <li>cargo-tauri</li>
    <li>cargo-tauri_1</li>
    <li>en-croissant</li>
    <li>gale</li>
    <li>insulator2</li>
    <li>mouse-actions-gui</li>
    <li>pot</li>
    <li>surrealist</li>
    <li>wealthfolio</li>
  </ul>
</details>

GitButler was previously failing. See #355157

File: pkgs/by-name/ca/cargo-tauri/test-app.nix
Comment (getchoo) [reply pkgs/by-name/ca/cargo-tauri/test-app.nix:30]: It's generally recommended to not use patchelf on binaries we're compiling from source anyways. This is also what basically every other Tauri package that requires the library does

**Comment by getchoo** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 346277`

---
### `x86_64-darwin`
<details>
  <summary>:fast_forward: 4 packages marked as broken and skipped:</summary>
  <ul>
    <li>alcom</li>
    <li>alexandria</li>
    <li>cinny-desktop</li>
    <li>insulator2</li>
  </ul>
</details>
<details>
  <summary>:x: 3 packages failed to build:</summary>
  <ul>
    <li>gitbutler</li>
    <li>modrinth-app</li>
    <li>modrinth-app-unwrapped</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>cargo-tauri</li>
    <li>cargo-tauri_1</li>
    <li>en-croissant</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:fast_forward: 4 packages marked as broken and skipped:</summary>
  <ul>
    <li>alcom</li>
    <li>alexandria</li>
    <li>cinny-desktop</li>
    <li>insulator2</li>
  </ul>
</details>
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>gitbutler</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>cargo-tauri</li>
    <li>cargo-tauri_1</li>
    <li>en-croissant</li>
    <li>modrinth-app</li>
    <li>modrinth-app-unwrapped</li>
  </ul>
</details>

modrinth-app fails on x86_64-darwin due to an existing build failure of `turbo`. gitbutler fails for the same reason as above

**Comment by Eveeifyeve** [comment]
> ## `nixpkgs-review` result
> Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).
> 
> Command: `nixpkgs-review pr 346277`
> 
> ### `x86_64-darwin`
> ⏩ 4 packages marked as broken and skipped:
> ❌ 3 packages failed to build:
> ✅ 3 packages built:
> ### `aarch64-darwin`
> ⏩ 4 packages marked as broken and skipped:
> ❌ 1 package failed to build:
> ✅ 5 packages built:
> modrinth-app fails on x86_64-darwin due to an existing build failure of `turbo`. gitbutler fails for the same reason as above

How did you do this?

**Comment by getchoo** [comment]
> an existing build failure of turbo

https://github.com/NixOS/nixpkgs/pull/355802

**Comment by Scrumplex** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 346277`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package marked as broken and skipped:</summary>
  <ul>
    <li>cinny-desktop</li>
  </ul>
</details>
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:x: 3 packages failed to build:</summary>
  <ul>
    <li>alcom</li>
    <li>alexandria</li>
    <li>gitbutler</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>cargo-tauri</li>
    <li>cargo-tauri_1</li>
    <li>en-croissant</li>
    <li>gale</li>
    <li>insulator2</li>
    <li>modrinth-app</li>
    <li>modrinth-app-unwrapped</li>
    <li>mouse-actions-gui</li>
    <li>pot</li>
    <li>surrealist</li>
    <li>treedome</li>
    <li>wealthfolio</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 4 packages marked as broken and skipped:</summary>
  <ul>
    <li>alcom</li>
    <li>cinny-desktop</li>
    <li>modrinth-app</li>
    <li>modrinth-app-unwrapped</li>
  </ul>
</details>
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:x: 8 packages failed to build:</summary>
  <ul>
    <li>en-croissant</li>
    <li>gale</li>
    <li>gitbutler</li>
    <li>insulator2</li>
    <li>mouse-actions-gui</li>
    <li>pot</li>
    <li>surrealist</li>
    <li>wealthfolio</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>cargo-tauri</li>
    <li>cargo-tauri_1</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:fast_forward: 4 packages marked as broken and skipped:</summary>
  <ul>
    <li>alcom</li>
    <li>alexandria</li>
    <li>cinny-desktop</li>
    <li>insulator2</li>
  </ul>
</details>
<details>
  <summary>:x: 3 packages failed to build:</summary>
  <ul>
    <li>gitbutler</li>
    <li>modrinth-app</li>
    <li>modrinth-app-unwrapped</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>cargo-tauri</li>
    <li>cargo-tauri_1</li>
    <li>en-croissant</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:fast_forward: 4 packages marked as broken and skipped:</summary>
  <ul>
    <li>alcom</li>
    <li>alexandria</li>
    <li>cinny-desktop</li>
    <li>insulator2</li>
  </ul>
</details>
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>gitbutler</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>cargo-tauri</li>
    <li>cargo-tauri_1</li>
    <li>en-croissant</li>
    <li>modrinth-app</li>
    <li>modrinth-app-unwrapped</li>
  </ul>
</details>

**Comment by Scrumplex** [comment]
Could you rebase and change `alcom` to use `cargo-tauri_1`?

**Comment by Scrumplex** [comment]
I will disregard gitbutler failures, as they will be fixed with #355366, modrinth-app seems to require a similar fix (not in this PR), the aarch64-linux failures are all reporting `failed to build app: Target aarch64-unknown-linux-gnu does not exist. Please run `rustup target list` to see the available targets.` which might be a regression as I can also see it on master.

**Comment by Scrumplex** [comment]
I would be inclined to merge this as is, but when building some of the aarch64-linux packages touched by this PR it fails for me with ``failed to build app: Target aarch64-unknown-linux-gnu does not exist. Please run `rustup target list` to see the available targets.``.

I don't have an actual arm builder. I use binfmt.emulatedSystems, so that might be the issue here?

**Comment by getchoo** [comment]
Seems so. This is the result when building on an Ampere server

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 346277`

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 3 packages marked as broken and skipped:</summary>
  <ul>
    <li>alcom</li>
    <li>modrinth-app</li>
    <li>modrinth-app-unwrapped</li>
  </ul>
</details>
<details>
  <summary>:fast_forward: 1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>gitbutler</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 9 packages built:</summary>
  <ul>
    <li>cargo-tauri</li>
    <li>cargo-tauri_1</li>
    <li>en-croissant</li>
    <li>gale</li>
    <li>insulator2</li>
    <li>mouse-actions-gui</li>
    <li>pot</li>
    <li>surrealist</li>
    <li>wealthfolio</li>
  </ul>
</details>

**Review by Scrumplex — APPROVED** [review]

**Comment by LordGrimmauld** [comment]
> It should be removed in 25.05 as it will no longer be maintained upstream

Is this still the plan?

**Comment by getchoo** [comment]
> Is this still the plan? I believe there is still tauri v1 apps in nixpkgs

Unless those packages can be updated or are also unmaintained, this might need to stay. Shouldn't be too bad though, as it seems upstream has continued to [sorta maintain](https://github.com/tauri-apps/tauri/releases/tag/tauri-v1.8.2) the 1.x series

> but those build tauri via cargo and don't depend on this, i believe

They use this CLI to build

An alternative could be to reimplement some of the wrapping this CLI does around Cargo, manual creation and installation of assets, etc., but it'd be a lot of work for little benefit IMO

**Comment by LordGrimmauld** [comment]
I went through all consumers of webkitgtk_4_0 in https://github.com/NixOS/nixpkgs/pull/398999

The current tauri v1 list:
```
/nix/store/0vl2aqw0jaxz9zzl5zvbwdlgk8xqvrxa-rquickshare-legacy-0.11.4.drv
/nix/store/189qb6x1afnhq11bgl6v9qhqfba6i7f0-quantframe-1.3.4.drv
/nix/store/3sjp8q7jiirz57knjn0bpfbsfmpjwm0x-cinny-desktop-4.6.0.drv
/nix/store/9f8di9mqqqr0s90b1kq686pxk2n29wyk-pot-3.0.6.drv
/nix/store/9q9bgb46p3dj6n9hzix5nl2448abf5wk-mouse-actions-gui-0.4.5.drv
/nix/store/bip47im0rnl2bk3wbjj6gzxkwg96xqgq-restic-browser-0.3.2.drv
/nix/store/cany9jpcr2i3vi5fm0qif3r9xdyz3gcp-alexandria-0.13.1.drv
/nix/store/fxzw2i8239f8zf9wx7al2cvzw9k5yd9v-insulator2-2.13.2.drv
/nix/store/h99ksdsnvs6kivgqfgkawhgcbv4j9miv-en-croissant-0.11.1.drv
```

Of those, i only really care about `cinny-desktop` and just a little about `en-croissant` (both do have some upstream maintenance). We probably do not want to break cinny on 25.05 so i guess this stays for now.

**Pull request merged**
